src/msg.c

```diff
@@ -587,8 +527,7 @@ static int8_t msg_auth(struct session *session,
 	hal_log_info("[session %p] Authenticating UUID: %s, TOKEN: %s",
 		     session, uuid, token);
 
-	result = proto_signin(proto_sock, uuid, token, property_changed,
-			      L_INT_TO_PTR(session->node_fd));
+	result = cloud_auth_device(mydevice->id, token);
```

Comment:
It will lead to a segmentation fault if mydevice is not found

Reply:
Ok